packages/node_modules/pouchdb-utils/src/rev.js

```diff
@@ -1,7 +1,15 @@
 import uuid from 'uuid';
+import { stringMd5 } from 'pouchdb-md5';
+import { clone } from 'pouchdb-utils';
 
-function rev() {
-  return uuid.v4().replace(/-/g, '').toLowerCase();
+function rev(doc, deterministic_rev) {
+  var clonedDoc = clone(doc);
```

Comment:
So I really want to reduce thte amount of cloning we do, plus this is quite fragile (there are other private fields that get set), we make the metadata vs data split futher down in parseDoc, it would be great to refactor so we could just md5(result.data) here instead of cloning

Reply:
We may want to punt on that for its patch though if it ends up too complicated

Reply:
Ok cool. Ideally we want to md5 the previous revision as well. I'll take a look into that.

Reply:
At this point, from what I can see. The doc normally looks something like this:

```js
{
    _id: '12342349b5904acaa7161f12a0fb31b59c'
   _rev: '1-456149b5904acaa7161f12a0fb31b59c'
   ...  other doc contents...
  __revtree: []
}
```
So for this pr I would like to do the clone and remove the _rev_tree. Then look at another PR that does the rev of the result.data